api/src/pcapi/core/offers/api.py

```diff
@@ -1455,3 +1457,37 @@ def update_offers_description_from_product_description(product_id: int) -> None:
             extra={"product": product_id, "offers": offer_ids, "exc": str(exception)},
         )
         raise exceptions.NotUpdateProductOrOffers(exception)
+
+
+def update_used_stock_price(stock: models.Stock, new_price: float) -> None:
+    if not stock.offer.isEvent:
+        raise ValueError("Only stocks associated with an event offer can be edited with used bookings")
+
+    stock.price = new_price
+    bookings_models.Booking.query.filter(
+        bookings_models.Booking.stockId == stock.id,
+    ).update({bookings_models.Booking.amount: new_price})
+
+    first_finance_event = (
+        finance_models.FinanceEvent.query.join(bookings_models.Booking, finance_models.FinanceEvent.booking)
+        .filter(
+            finance_models.FinanceEvent.status != finance_models.FinanceEventStatus.CANCELLED,
+            bookings_models.Booking.stockId == stock.id,
+        )
+        .order_by(
+            finance_models.FinanceEvent.pricingOrderingDate,
+            finance_models.FinanceEvent.id,
+        )
+        .options(
+            sa.orm.joinedload(finance_models.FinanceEvent.booking),
+        )
+        .one_or_none()
```

Comment:
Cette requête récupère le FinanceEvent de **toutes** les réservations non annulées et liées au stock modifié. Il peut y avoir **plusieurs** réservations, et donc **plusieurs** FinanceEvent. Il faut donc utiliser `first()` : 

```suggestion
        .first()
```

Reply:
en l'absence de @Meewan , je viens de corriger (+ rebase et résolution de conflits)